lib/kennel.rb

```diff
@@ -100,26 +101,30 @@ def parts_serializer
 
     def generated
       @generated ||= begin
-        Progress.progress "Generating" do
+        parts = Progress.progress "Finding applicable projects and parts" do
           projects = projects_provider.projects
           projects = filter.filter_projects projects
 
           parts = Utils.parallel(projects, &:validated_parts).flatten(1)
-          parts = filter.filter_parts parts
+          filter.filter_parts parts
```

Comment:
don't we need to re-assign here since filter does not modify ?

Reply:
That's done by `Progress.progress` returning the value of the block